artcommon/artcommonlib/konflux/package_rpm_finder.py

```diff
@@ -86,7 +86,7 @@ def get_installed_packages_build_dicts(self, build_record: KonfluxBuildRecord) -
         installed_packages = [pkg for pkg in installed_packages if pkg not in self._not_found_packages]
         self._cache_packages(installed_packages)
         return {self._packages_build_dicts[package_nvr]['name']: self._packages_build_dicts[package_nvr]
-                for package_nvr in installed_packages}
+                for package_nvr in installed_packages if package_nvr not in self._not_found_packages}
```

Comment:
I think the if condition is repeated here? (already checked at line 86?)

Reply:
I think it's intended. First `installed_packages` is subtracted the packages we already know are not found. Then, we cache the rest. There can be more that are not mapped in Brew, so this condition again ensures we're not raising a KeyError by trying to index a package that could not be mapped.